### PR TITLE
Added Missing Audit and Failed Data Record while processing Rch xml

### PR DESCRIPTION
**Pull request by atish160384**
Added Audit record, Failed Data Record for Rch Xml's Deserialization Failed, RECORD Field Missing in xml and Null pointer Exception.

### CODE REVIEW

File: rch/src/main/java/org/motechproject/nms/rch/service/impl/RchWebServiceFacadeImpl.java [hunk 1]
```diff
@@ -1309,18 +1357,25 @@ private void healthblockFileProcess(String fileName, String result, Long stateId
...
+                rchImportFailRecordDataService.create(new RchImportFailRecord(endDate, RchUserType.HEALTHBLOCK, stateId));
             }
         } catch (JAXBException e) {
-            throw new RchInvalidResponseStructureException(String.format("Cannot deserialize RCH mother data from %s location.", stateId), e);
+            String error = String.format("Cannot deserialize RCH mother data from %s location.", stateId);
+            rchImportAuditDataService.create(new RchImportAudit(startDate, endDate, RchUserType.HEALTHBLOCK, stateCode, stateName, 0, 0, error));
+            rchImportFailRecordDataService.create(new RchImportFailRecord(endDate, RchUserType.HEALTHBLOCK, stateId));
+            throw new RchInvalidResponseStructureException(error, e);
         } catch (RchInvalidResponseStructureException e) {
             String error = String.format("Cannot read RCH mothers data from %s state with stateId: %d. Response Deserialization Error", stateName, stateId);
```
Comment (VishnuVimal) [inline rch/src/main/java/org/motechproject/nms/rch/service/impl/RchWebServiceFacadeImpl.java]: The error comments still have mothers in this method. Need to be changed

**Comment by atish160384** [comment]
Already merged

File: rch/src/main/java/org/motechproject/nms/rch/service/impl/RchWebServiceFacadeImpl.java [hunk 2]
```diff
@@ -519,7 +531,9 @@ private void districtFileProcess(String fileName, String result, Long stateId, S
             rchImportAuditDataService.create(new RchImportAudit(startDate, endDate, RchUserType.DISTRICT, stateCode, stateName, 0, 0, error));
             rchImportFailRecordDataService.create(new RchImportFailRecord(endDate, RchUserType.DISTRICT, stateId));
         } catch (NullPointerException e) {
```
Comment (gudipatiharitha) [inline rch/src/main/java/org/motechproject/nms/rch/service/impl/RchWebServiceFacadeImpl.java]: Shall we change all NullPointerExce to just Exception and "No files saved a" to "No files saved TYPE + e.getMessage () "

Comment (atish160384) [reply rch/src/main/java/org/motechproject/nms/rch/service/impl/RchWebServiceFacadeImpl.java]: ok

Comment (atish160384) [reply rch/src/main/java/org/motechproject/nms/rch/service/impl/RchWebServiceFacadeImpl.java]: Lets go ahead with exception

**Pull request merged**